File: PrimaryArithmetic.java

import java.util.Scanner;
public class PrimaryArithmetic {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int a = sc.nextInt(), b = sc.nextInt();
            if (a == 0 && b == 0) break;
            int carry = 0, count = 0;
            while (a > 0 || b > 0) {
                if ((a % 10 + b % 10 + carry) >= 10) {
                    carry = 1;
                    count++;
                } else {
                    carry = 0;
                }
                a /= 10;
                b /= 10;
            }
            if (count == 0) System.out.println("No carry operation.");
            else if (count == 1) System.out.println("1 carry operation.");
            else System.out.println(count + " carry operations.");
        }
    }
}

File: primaryarithmetic.py
while True:
    a, b = map(int, input().split())
    if a == 0 and b == 0:
        break

    carry = 0
    count = 0

    while a > 0 or b > 0:
        d1 = a % 10
        d2 = b % 10
        if d1 + d2 + carry >= 10:
            count += 1
            carry = 1
        else:
            carry = 0
        a //= 10
        b //= 10

    if count == 0:
        print("No carry operation.")
    elif count == 1:
        print("1 carry operation.")
    else:
        print(f"{count} carry operations.")
